### DISCUSSION TITLE
Gatsby Cloud + Drupal - publishing is faster than previews

### DISCUSSION THREAD

**Discussion by hughevans** [gatsbyjs/gatsby#30341] [Help]
Hi, just trying to work out why when I make an update in Drupal it does a data push to Gatsby Cloud and deploys in a few seconds, but it takes much longer for the preview to update (if at all).

<img width="1003" alt="Screen Shot 2021-03-19 at 11 08 39 am" src="https://user-images.githubusercontent.com/2580/111713600-11d33300-88a4-11eb-9729-63c4c0aa4c58.png">
<img width="994" alt="Screen Shot 2021-03-19 at 11 11 40 am" src="https://user-images.githubusercontent.com/2580/111713608-1566ba00-88a4-11eb-8072-5b52f430375c.png">
<img width="809" alt="Screen Shot 2021-03-19 at 11 14 25 am" src="https://user-images.githubusercontent.com/2580/111713838-7b534180-88a4-11eb-8026-5395bc04fe97.png">

I’m now using the `gatsby_endpoints` module, but the previews issue existed prior to that also. 

It’s like there is no push event to the preview server, but there is to the build server. Is that expected?

**Comment by hughevans** [comment]
I’m still stuck with this still. Can anyone assist? I’m not even sure if the hook back to the preview server is being triggered (the build one is though). I might modify the Drupal plugin and then log out its calls.

**Reply by vladar** [reply]
Hey @hughevans Could you contact our cloud support via https://www.gatsbyjs.com/support/ ? It is a better channel to tackle problems like this.
